src/triggers/newCandyMachineNfts.ts

```diff
@@ -0,0 +1,107 @@
+
+
+import { Metadata } from '@metaplex-foundation/js';
+import { web3 } from '@project-serum/anchor';
+import { bs58 } from '@project-serum/anchor/dist/cjs/utils/bytes';
+import { ConfirmedSignatureInfo, PublicKey } from '@solana/web3.js';
+import _ from 'lodash';
+
+import { Table } from '../db/db';
+import { MetaFactory } from '../types/metaFactory';
+import { NftFactory } from '../types/nft';
+import { Clients } from '../types/processor';
+import { Trigger } from '../types/trigger';
+
+const MAX_NAME_LENGTH = 32;
+const MAX_URI_LENGTH = 200;
+const MAX_SYMBOL_LENGTH = 10;
+const MAX_CREATOR_LEN = 32 + 1 + 1;
+const MAX_CREATOR_LIMIT = 5;
+const MAX_DATA_SIZE = 4 + MAX_NAME_LENGTH + 4 + MAX_SYMBOL_LENGTH + 4 + MAX_URI_LENGTH + 2 + 1 + 4 + MAX_CREATOR_LIMIT * MAX_CREATOR_LEN;
+const MAX_METADATA_LEN = 1 + 32 + 32 + MAX_DATA_SIZE + 1 + 1 + 9 + 172;
+const CREATOR_ARRAY_START = 1 + 32 + 32 + 4 + MAX_NAME_LENGTH + 4 + MAX_URI_LENGTH + 4 + MAX_SYMBOL_LENGTH + 2 + 1 + 4;
+
+const TOKEN_METADATA_PROGRAM = new PublicKey('metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s');
+const CANDY_MACHINE_V2_PROGRAM = new PublicKey('cndy3Z4yapfJBmL3ShUp5exZKqR3z33thTzeNMm2gRZ');
```

Comment:
should we put this and the generic candy machine methods in a candy machine client instead of in the trigger? or i guess if the only place they're likely to be used is in this trigger then it's fine to leave here